docs/aws/audit/sgaudit/rules/vpc_exists_without_flow_logs.mdx

```diff
@@ -23,6 +23,280 @@ HIPAA, PCIDSS, GDPR, SOC2, CISAWS, CBP
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the Amazon VPC console at https://console.aws.amazon.com/vpc/.
+
+2. In the navigation pane, choose 'Your VPCs'.
+
+3. Select the VPC for which you want to check the flow logs. 
+
+4. In the details pane, on the 'Flow logs' tab, check if there are any flow logs enabled for the selected VPC. If there are no flow logs or if they are not properly configured, it indicates a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the VPCs in your account using the following command:
+   ```
+   aws ec2 describe-vpcs
+   ```
+   This command will return a JSON output with details of all the VPCs.
+
+3. To check if flow logs are enabled for a specific VPC, you need to use the following command:
+   ```
+   aws ec2 describe-flow-logs --filter Name=resource-id,Values=<vpc-id>
+   ```
+   Replace `<vpc-id>` with the ID of the VPC you want to check. This command will return a JSON output with details of the flow logs. If the output is empty, it means that flow logs are not enabled for the VPC.
+
+4. To check flow logs for all VPCs, you can use a script to loop through all the VPCs and check if flow logs are enabled. Here is a simple script in bash:
+   ```
+   for vpc in $(aws ec2 describe-vpcs --query 'Vpcs[].VpcId' --output text)
+   do
+     echo "Checking VPC: $vpc"
+     aws ec2 describe-flow-logs --filter Name=resource-id,Values=$vpc
+   done
+   ```
+   This script will print the details of the flow logs for each VPC. If the output for a VPC is empty, it means that flow logs are not enabled for that VPC.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install Boto3 in your Python environment. You can do this using pip:
+   ```
+   pip install boto3
+   ```
+   After installing Boto3, you need to configure it with your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   ```
+
+2. Use Boto3 to list all VPCs:
+   You can use the `describe_vpcs` method of the EC2 client in Boto3 to list all VPCs. Here is a sample script:
+   ```python
+   import boto3
+
+   ec2 = boto3.client('ec2')
+
+   response = ec2.describe_vpcs()
+
+   for vpc in response['Vpcs']:
+       print(vpc['VpcId'])
+   ```
+
+3. Use Boto3 to check if flow logs are enabled for each VPC:
+   You can use the `describe_flow_logs` method of the EC2 client in Boto3 to check if flow logs are enabled for each VPC. Here is a sample script:
+   ```python
+   import boto3
+
+   ec2 = boto3.client('ec2')
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>